pkg/grafana/grafana.go

```diff
@@ -24,13 +24,14 @@ import (
 	"strings"
 )
 
-type DashboardsInterface interface {
-	Search() ([]GrafanaDashboard, error)
-	Create(dashboardJson io.Reader) error
-	Delete(slug string) error
+type APIInterface interface {
+	SearchDashboard() ([]GrafanaDashboard, error)
+	CreateDashboard(dashboardJson io.Reader) error
+	DeleteDashboard(slug string) error
+	CreateDatasource(datasourceJson io.Reader) error
 }
 
-type DashboardsClient struct {
+type APIClient struct {
```

Comment:
You can do something like

```golang
type APIClient struct {
  *httpClient
  BaseUrl *url.URL
}
```

This is functionally similar to inheritance, in that an instance of `APIClient` can call methods on `http.Client` without using `apiClient.HTTPClient.doPost`

Reply:
Nice! I read up on this too.